fix multiple host emails bugs

diegocarhuavilca opened the pull request.

diegocarhuavilca committed:
feat: Update hosts and scans mockup

diegocarhuavilca committed:
feat: Update hosts and scans mockup

diegocarhuavilca committed:
fix multiple host emails bugs

grez-lucas commented on src/components/Form/FormHostEmails.vue:
```diff
@@ -84,23 +85,46 @@
-      is_principal: false
-    });
+
+    if (hostsRegister.value[index].alias === 'All') {
```
I get an error when adding a rapporteur to two hosts on this line:

![image](https://github.com/user-attachments/assets/5ce8b412-1e4a-47a3-b2d3-a3ffc1791215)

grez-lucas reviewed (changes requested):
Good fixes! Please review the aforementioned bug, and make your PR target branch `dev` instead of `staging` 👍🏻

diegocarhuavilca committed:
feat : Validation should have a green notification when successful

Export HOST filtering components to Header

Hosts table Email column must show principal email associated with the host

grez-lucas commented on src/components/Form/FormHostEmails.vue:
A Host should only have (at most) **one** `is_principal` **Rapporteur**. Please implement a method so that clicking on the `fa-heart` button follows this logic. That is - unsetting other is_principal values from other rapporteurs and setting that rapporteur to be the _only_ one who `is_principal` 👍🏻  

![image](https://github.com/user-attachments/assets/73fcf2f4-08ff-4868-8822-cd6d4db2cad0)

grez-lucas commented on src/components/Table/TableRegular.vue:
I've added a [bug task in Jira ](https://deltateam.atlassian.net/jira/software/projects/KT/boards/2?isInsightsOpen=true&selectedIssue=KT-258) for a pagination bug that happens in this component. Having ~11 hosts, my "actions" column begins to multiply:

### 11 Hosts, 5 per page, page 1:
![image](https://github.com/user-attachments/assets/4e322a51-a809-4f92-9348-a5c0836ffd75)

### 11 Hosts, 5 per page, page 2:
![image](https://github.com/user-attachments/assets/96b26cc0-a745-40cb-8002-de40a7ba4b6b)


### 11 Hosts, 10 per page, page 1:
![image](https://github.com/user-attachments/assets/893a06eb-a84b-4d73-8635-f9c4f16df060)


### 21 Hosts, All per page, page 1:

- Hosts overflow and the user is not able to scroll to see them properly
![image](https://github.com/user-attachments/assets/df939d4e-971d-4241-b549-880a9c940591)

diegocarhuavilca replied:
@grez-lucas It's ready to test also with the jira tasks

grez-lucas replied:
Thanks! I'll give it a check briefly 👍🏻

grez-lucas replied:
Don't forget to push your commits so I can review them please!

grez-lucas reviewed (changes requested):
Nice additions! 

- Validation green notif ✅
- Host filtering on header ✅
- Adding rapporteur to all ✅


Pending:
- Fix only one principal per host logic 💫
- Fix actions column duplicating on pagination 💫
- Fix overflow error on ~20+ hosts per table when showing "All"

diegocarhuavilca committed:
feat: Fix multiple ui bugs for host

grez-lucas reviewed (approved):
Good work solving these bugs! 👍🏻
